datalad/distribution/update.py

```diff
@@ -79,9 +103,33 @@ class Update(Interface):
             constraints=EnsureDataset() | EnsureNone()),
         merge=Parameter(
             args=("--merge",),
-            action="store_true",
-            doc="""merge obtained changes from the given or the
-            default sibling""", ),
+            metavar="ALLOWED",
+            # const and nargs are set to map --merge to --merge=any.
+            const="any",
+            nargs="?",
+            constraints=EnsureBool() | EnsureChoice("any", "ff-only"),
+            doc="""merge obtained changes from the sibling. If a sibling is not
+            explicitly given and there is only a single known sibling, that
+            sibling is used. Otherwise, an unspecified sibling defaults to the
+            configured remote for the current branch. By default, changes are
+            fetched from the sibling but not merged into the current branch.
+            With [CMD: --merge or --merge=any CMD][PY: merge=True or
+            merge="any" PY], the changes will merged into the current branch. A
+            value of 'ff-only' restricts the allowed merges to
+            fast-forwards."""),
+        follow=Parameter(
+            args=("--follow",),
+            constraints=EnsureChoice("sibling", "parent"),
+            doc="""source of updates for subdatasets. For 'sibling', the update
+            will be done by merging in a branch from the (specified or
+            inferred) sibling. The branch brought in will either be the current
+            branch's configured branch, if it points to a branch that belongs
+            to the sibling, or a branch on the sibling branch with a name that
+            matches the current branch. For 'parent', the revision registered
+            in the parent of the subdataset is merged in. Note that the current
```

Comment:
I think I remembered why I also disliked "parent". "Parent" is a term you see in git to describe "parent commit".  So, `--follow=parent` could be understood (if not reading this doc) pretty-much as `--ff-only`.
I think this notion of "parent" been a super-dataset is also too implicit in the above sentence in "parent of the subdataset", especially if we consider that "parent" could indeed be a "parent commit".
So if we are to stay with "parent" (I also have dislike of "superdataset" since it is not just "an immediate superdataset"), doc might better be adjusted ".. in the superdataset, immediate parent of the subdataset, ..." .  A bit mouthful but removes that "implicit" meaning.

But may be, it should become `parent-superdataset` or `parent-dataset` to be 100% explicit in its meaning?

Reply:
> I think I remembered why I also disliked "parent".

I'm left wondering what the first thing you disliked was :]

> "Parent" is a term you see in git to describe "parent commit".

I see.  I agree that a parent is a very context-dependent thing, and the question is "parent of what?".  You have the parent (dataset) of the dataset, parent (commit or commits) of the commit, parent (directory) of a directory...  So yeah, perhaps it's best to stay away from an option value that is open to confusion.

(All of the above can apply to "sibling" too, but there's less potential for confusion because user-facing git documentation doesn't use the term currently.)

> So, `--follow=parent` could be understood (if not reading this doc) pretty-much as `--ff-only`.

Hrm, I was with you on `--follow=parent` leading to a user being not quite sure what "parent" was referring to, but you've lost me here.  Merging the parent commit is always a no-op.  I'm not seeing how a user would get to the `--ff-only` conclusion.

> I think this notion of "parent" been a super-dataset is also too implicit in the above sentence in "parent of the subdataset", especially if we consider that "parent" could indeed be a "parent commit".

What would be the "parent commit of a subdataset"?  I'm of course biased because I wrote the sentence in question, but I think the natural and clear reading of "parent of X" is that the parent of the same kind (here "dataset") of X.

> So if we are to stay with "parent" (I also have dislike of "superdataset" since it is not just "an immediate superdataset"), doc might better be adjusted ".. in the superdataset, immediate parent of the subdataset, ..." . A bit mouthful but removes that "implicit" meaning.

My view is that you're overthinking this.  While it's of course best to remove ambiguity, I think "parent of X" can safely be assumed to be first-level.  Otherwise, the writer would say it.  As an example, I haven't heard of any Git users being confused by "parent commit", but git doesn't ever qualify parent with "immediate".  (OK, there's one code comment in revision.c that uses the term :].)

> But may be, it should become `parent-superdataset` or `parent-dataset` to be 100% explicit in its meaning?

Those are fine by me, with a preference toward "parent-dataset".  I like that it (almost) matches the key that `datalad-subdatasets` uses, "parentds".  I could be persuaded to use that directly, though I think it makes sense to avoid an unnecessary abbreviation here.

Any other thoughts on what this option value should be?

Reply:
@adswa It'd be nice to hear your thoughts on this too.  That way I'm not to blame when you're struggling to explain a confusingly named option in the handbook :]

Reply:
to keep it short: I like `parentds` since it is already used with the same meaning and explicit.